server/src/protocol-translation.ts

```diff
@@ -233,7 +238,15 @@ export function asTagsDocumentation(tags: tsp.JSDocTagInfo[]): string {
 export function asTagDocumentation(tag: tsp.JSDocTagInfo): string {
     switch (tag.name) {
         case 'param': {
-            const body = (tag.text || '').split(/^([\w.]+)\s*-?\s*/);
+            // TODO: What does this actually look like? Is the original logic
+            // just doing something that the new type should do?
+            let formattedTag = '';
+            if (typeof tag.text === 'string') {
+                formattedTag = tag.text;
+            } else if (tag.text) {
+                formattedTag = tag.text.map(symbol => `${symbol.text} (${symbol.kind})`).join(', ');
+            }
```

Comment:
I'd like something like:
```suggestion
const formattedTag = typeof tag.text === 'string' ? tag.text : asPlainText(tag.text);
```

and then update `asPlainText` to handle the `kind` in `SymbolDisplayPart`.

Reply:
But to be fair, looking at the description of `kind` which is:

> The symbol's kind (such as 'className' or 'parameterName' or plain 'text')

I don't think we would necessarily want to to show the kind value. Certainly not when it's `text` but even in other cases it's probably not needed (at least for now).

So it seems fine to just do:

```
const formattedTag = typeof tag.text === 'string' ? tag.text : asPlainText(tag.text);
```
for now without updating `asPlainText`.

Reply:
Okay, I'm probably fine with that. I'm still playing with this, I want to see what it ends up looking like in https://github.com/apexskier/nova-typescript/ with full information.

Reply:
Cool, please let me not (screenshots would be nice :)).

BTW. To enable typescript to send `SymbolDisplayPart` I guess some flag/setting needs to be enabled?

Reply:
Though note the documentation that I've referenced above:

> The symbol's kind (such as 'className' or 'parameterName' or plain 'text')

Those kinds look too low-level to show in the UI anyway. If we would want to show those we probably would have to map them to UI-friendly strings.

Reply:
Looks like it's controlled by `UserPreferences.displayPartsForJSDoc`
https://github.com/microsoft/TypeScript/blob/62f9155a9d3c734854f3340e2b74fb799bf407fe/src/server/protocol.ts#L989

Reply:
At time of writing with the TODO experimental code, here's an example of what this looks like (obviously lots of work to be done)

<img width="566" alt="TypeScript Extension – index d ts - Nova - 2021-07-02 at 21 34 45" src="https://user-images.githubusercontent.com/329222/124321078-5ba70700-db7d-11eb-8a06-b40c63a4c701.png">
<img width="627" alt="TypeScript Extension – novaUtils ts - Nova - 2021-07-02 at 21 34 58" src="https://user-images.githubusercontent.com/329222/124321079-5c3f9d80-db7d-11eb-9758-77e0a1f8d53e.png">

Reply:
Right. So that `text` kind looks very redundant so far. I wonder in which cases we would get the other "kind"s.

Reply:
FYI, I'm removing this in this initial PR. I'll do a secondary one to change the formatting behavior.